tests/improc/test_zogy.py

```diff
@@ -0,0 +1,436 @@
+                sim.stars.star_mean_x_pos[i] += 15  # bump this star out of the way
+            if abs(p - sim.stars.star_mean_y_pos[i]) < 10:
+                sim.stars.star_mean_y_pos[i] += 15  # bump this star out of the way
+
+    # beware the 4-loop!
+    for ref_seeing in seeing_values:
+        for new_seeing in seeing_values:
+            for ref_bkg in background_values:
+                for new_bkg in background_values:
+                    # print(f'seeing: {ref_seeing}, {new_seeing} | bkg: {ref_bkg}, {new_bkg}')
+
+                    sim.pars.seeing_mean = ref_seeing
+                    sim.pars.background_mean = ref_bkg
+
+                    # remove the additional stars from the reference image
+                    sim.stars.remove_stars(len(sim.stars.star_mean_fluxes) - sim.pars.star_number)
+
+                    sim.make_image(new_sky=True)
+                    truth1 = sim.truth
+                    im1 = sim.apply_bias_correction(sim.image)
+                    im1 -= truth1.background_instance
+                    psf1 = truth1.psf_downsampled
+                    bkg1 = truth1.total_bkg_var
+
+                    # add a few sources
+                    for f, p in zip(fluxes, pos):
+                        sim.add_extra_stars(flux=f, x=p, y=p)
+
+                    sim.pars.seeing_mean = new_seeing
+                    sim.pars.background_mean = new_bkg
+
+                    sim.make_image(new_sky=True)
+                    truth2 = sim.truth
+                    im2 = sim.apply_bias_correction(sim.image)
+                    im2 -= truth2.background_instance
+                    psf2 = truth2.psf_downsampled
+                    bkg2 = truth2.total_bkg_var
+
+                    # need to figure out better values for this
+                    F1 = 1.0
+                    F2 = 1.0
+
+                    zogy_diff, zogy_psf, zogy_score, zogy_score_corr, alpha, alpha_err = zogy_subtract(
+                        im1, im2, psf1, psf2, np.sqrt(bkg1), np.sqrt(bkg2), F1, F2,
+                    )
+
+                    edge = 8  # do not trigger on stars too close to the edge
+                    S = np.ones(zogy_score_corr.shape) * np.nan
+                    S[edge:-edge, edge:-edge] = zogy_score_corr[edge:-edge, edge:-edge].copy()
+                    B = np.sqrt(ref_bkg ** 2 + new_bkg ** 2)
+                    P = zogy_psf
+
+                    for i, f in enumerate(fluxes):
+                        x = y = int(pos[i])
+                        c = S[y - edge:y + edge, x - edge:x + edge]
+                        expected = f * np.sqrt(np.sum(P ** 2) / B)
+                        expected *= np.sqrt(B / (B + f * np.sum(P ** 2)))  # add the Poisson noise from the source
+                        measured = np.nanmax(c)
+                        # print((expected - measured) / expected)
+                        # TODO: figure out why we still have cases where the S/N is reduced by 50%
```

Comment:
What does the histogram of S/N expected vs. measured look like?

Reply:
![zogy_snr_histograms](https://github.com/c3-time-domain/SeeChange/assets/37179063/79f1386f-bea8-4ee6-8451-8a51e019064f)

Reply:
The distributions look ok, but there's definitely a mismatch of 15% loss of S/N for fluxes above the background. 
I will try to figure this out, but I think we should merge and keep at it on another PR.